src/test/java/com/ericsson/ei/notifications/EmailSenderTest.java

```diff
@@ -51,9 +51,9 @@ public void sendEmailThrowsException() throws Exception {
         emailSender.sendEmail(message);
     }
 
-    @Test
+    @Test(expected = NullPointerException.class)
     public void testDefaultValues() {
-        assertEquals("noreply@ericsson.com", emailSender.getSender());
+        assertEquals("noreply@domain.com", emailSender.getSender());
         assertEquals("Email Subscription Notification", emailSender.getSubject());
```

Comment:
Shouldn't this test call upon emailSender.prepareEmailMessage? Then we assert the values of subject and sender in the created message? If we send in a null value for this parameter in the test, we expect the default values to have been added. And I don't think there is a need for the expected NullPointerException then.

Reply:
yah I have tried putting upon emailSender.prepareEmailMessage but still facing the error.
Actually in EmailSenderTest class its mentioned previously javadocs that 
 /*
     * prepareEmail is not possible to test due too much hidden Spring mumbo jumbo doing things in
     * background.
     */
so I think should remove the testcase.

Reply:
I took a look at the code again, and it seems very difficult to mock the MimeMessage and MimeMessageHelper... Since the default values doesn't get added until we actually use the prepareEmail method this test is rendered useless. I say remove this unit test and instead keep the cucumber functional test which actually sends an email with the correct subject + sender...

Reply:
I agree with you, remove this test case then.